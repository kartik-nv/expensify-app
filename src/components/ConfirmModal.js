import React from 'react'
import Modal from 'react-modal'

const ConfirmModal = (props) => (
    <Modal
        isOpen={props.isModalVisible}
        contentLabel="Confirm Remove Expense"
        onRequestClose={props.handleModalClose}
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Warning</h3>
        <p className="modal__body">Are you sure to remove this expense?</p>
        <div className="modal__buttons">
            <button className="button" onClick={props.removeExpense}>Remove</button>
            <button className="button button--modal" onClick={props.handleModalClose}>Cancel</button>
        </div>
        
    </Modal>
)

export default ConfirmModal