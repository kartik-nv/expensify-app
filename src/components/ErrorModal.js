import React from 'react'
import Modal from 'react-modal'

const ErrorModal = (props) => (
    <Modal
        isOpen={!!props.isModalVisible}
        contentLabel="Login Error"
        onRequestClose={props.handleModalClose}        
        closeTimeoutMS={200}
    className="modal"
    ariaHideApp={false}
    >
        <h3 className="modal__title">Error</h3>
    <p className="modal__body">{ `Your account with ${props.isModalVisible} already exists. Please Login via ${props.isModalVisible}.` }</p>
        <button className="button button--modal" onClick={props.handleModalClose}>OK</button>        
    </Modal>
)

export default ErrorModal