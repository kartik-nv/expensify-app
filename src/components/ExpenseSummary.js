import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpensesList from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpenseSummary = (props) => {
    return (
        <div>
            {props.expenseCount > 1 ? (<p>{`Viewing ${props.expenseCount} expenses totalling Rs.${numeral(props.expenseTotal).format('0.00')}`}</p>) : (<p>{`Viewing ${props.expenseCount} expense totalling Rs.${numeral(props.expenseTotal).format('0.00')}`}</p>)}
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpensesList(state.expenses, state.filters).length,
        expenseTotal: selectExpensesTotal(selectExpensesList(state.expenses, state.filters))
    }
}

export default connect(mapStateToProps)(ExpenseSummary)