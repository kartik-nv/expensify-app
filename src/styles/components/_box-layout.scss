.box-layout {
  background: url('/images/bg.jpg');
  background-size: cover;
  display: flex;
  justify-content: center;
  align-items: center;
  background-repeat: no-repeat;
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(white, .20);
  border-radius: 5px;
  padding: $l-size $l-size;
  text-align: center;
  width: 40rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
