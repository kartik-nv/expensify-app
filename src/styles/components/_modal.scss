.ReactModalPortal > div {
  opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay {
  align-items: center;
  display: flex;
  justify-content: center;
  transition: opacity 200ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
  opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
  opacity: 0;
}

.modal {
  background: #f2f2f2;
  color: black;
  max-width: 30rem;
  outline: none;
  padding: $m-size;
  text-align: center;
  border-radius: 5px;
  @media (min-width: $desktop-breakpoint) {
      max-width: 40rem;
      padding: $l-size;
  }
}

.modal__title {
  margin: 0 0 $m-size 0;
  border-bottom: 1px solid #dbd9d9;
  @media (min-width: $desktop-breakpoint) {
    margin: 0 0 $l-size, 0;
    font-size: $l-size;
  }
}

.modal__body {
  font-size: 1.8rem;
  font-weight: 200;
  margin: 0 0 $s-size 0;
  overflow-wrap: break-word;
  @media (min-width: $desktop-breakpoint) {
    font-size: 2rem;
    font-weight: 300;
    margin: 0 0 $l-size 0;
  }
}

.modal__buttons {
  display: flex;
  justify-content: space-between;
  margin: 0 $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin: 0 $l-size;
  }
}