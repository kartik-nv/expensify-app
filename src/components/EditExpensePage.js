import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal'
import ExpenseForm from './ExpenseForm';
import ConfirmModal from './ConfirmModal';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {

  state = {
    isModalVisible: false
  }

  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  componentWillMount() {
    Modal.setAppElement('body')
  }

  handleModalClose = () => {
    this.setState(() => ({
      isModalVisible: false
    }))
  }

  handleModalOpen = () => {
    this.setState(() => ({
      isModalVisible: true
    }))
  }

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.handleModalOpen}>Remove Expense</button>
        </div>
        <ConfirmModal 
          isModalVisible={this.state.isModalVisible}
          handleModalClose={this.handleModalClose}
          removeExpense = {this.onRemove}
        />
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  removeExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
