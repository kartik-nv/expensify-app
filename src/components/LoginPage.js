import React from 'react';
import { connect } from 'react-redux';
import { startLoginWithGoogle, startLoginWithFacebook } from '../actions/auth';
import ErrorModal from './ErrorModal';

export class LoginPage extends React.Component {

  state = {
    isError: undefined
  }

  FacebookAuth = () => {
    this.props.startLoginWithFacebook().catch(() => {
      this.setState(() => ({
        isError: 'Google'
      }))
    })
  }

  GoogleAuth = () => {
    this.props.startLoginWithGoogle().catch(() => {
      this.setState(() => ({
        isError: 'Facebook'
      }))
    })
  }

  handleModalClose = () => {
    this.setState(() => ({
      isError: undefined
    }))
  }

  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">Expensify</h1>
          <p>It's time to track your expenses without a hitch.</p>
          <div className="box-layout__login-buttons">
            <button
              className="button"
              onClick={this.GoogleAuth}
              >
              Login with Google
            </button>
            <p>or</p>
            <button
              className="button"
              onClick={this.FacebookAuth}
            >
              Login with Facebook
            </button>
          </div>
        </div>
        <ErrorModal isModalVisible={this.state.isError} handleModalClose={ this.handleModalClose } />
        
      </div>
    )
  }
} 

const mapDispatchToProps = (dispatch) => ({
  startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
  startLoginWithFacebook: () => dispatch(startLoginWithFacebook())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
