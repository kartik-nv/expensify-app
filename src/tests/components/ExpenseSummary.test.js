import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render ExpenseSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={3600} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={5} expenseTotal={36004578924} />)
    expect(wrapper).toMatchSnapshot()
})