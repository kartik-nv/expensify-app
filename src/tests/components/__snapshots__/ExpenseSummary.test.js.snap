// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        Rs. 
        0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        5
      </span>
       
      expenses
       totalling 
      <span>
        Rs. 
        0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
